File: ios-sdk/Sources/TezosMobileSDK/Beacon/DeepLinkHandler.swift
import Foundation

public enum BeaconDeepLinkHandler {
    public static func parsePairing(from url: URL) -> BeaconPairingRequest? {
        guard url.host == "beacon", url.path == "/pair" else { return nil }
        guard let components = URLComponents(url: url, resolvingAgainstBaseURL: false) else { return nil }
        guard let dataItem = components.queryItems?.first(where: { $0.name == "data" })?.value else { return nil }
        return try? BeaconCodec.decodePairingRequest(fromBase64: dataItem)
    }
}

File: android-sdk/src/main/java/io/tezos/mobile/beacon/DeepLinkHandler.kt
package io.tezos.mobile.beacon

import android.net.Uri

object DeepLinkHandler {
    fun parsePairing(uri: Uri): BeaconPairingRequest? {
        if (uri.host != "beacon" || uri.path != "/pair") return null
        val data = uri.getQueryParameter("data") ?: return null
        return Codec.decodePairingRequest(data)
    }
}


